slurmutils/editors/editor.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Base methods for Slurm workload manager configuration file editors."""
+
+import logging
+import shlex
+from functools import wraps
+from os import path
+from typing import Any, Dict, List, Tuple
+
+from ..exceptions import EditorError
+
+_logger = logging.getLogger("slurmutils")
+
+
+def _is_comment(line: str) -> bool:
+    """Check if line is a comment."""
+    return line.startswith("#")
+
+
+def _contains_comment(line: str) -> bool:
+    """Check if line contains an inline comment."""
+    return "#" in line
+
+
+def _slice_comment(line: str) -> str:
+    """Slice inline comment off of line."""
+    return line.split("#", maxsplit=1)[0]
+
+
+def clean(line: str) -> Tuple[str, bool]:
+    """Clean line before further processing.
+
+    Returns:
+        Returns the cleaned line and False if it should be ignored.
+        If True, then the processors should ignore the line.
+    """
+    if _is_comment(line):
+        return "", True
+
+    return (_slice_comment(line) if _contains_comment(line) else line).strip(), False
```

Comment:
Now that I read this again, does this need all the guards? I'm seeing that `_slice_comment` will already return the correct result in all cases, even if the line starts with "#" or if the line has a trailing comment.

Reply:
Hmm... I originally used `_is_comment(...)` to quickly check if a line is comment and we shouldn't worry about even performing the slice. I can reduce this implementation to see if `_is_comment(...)` is really necessary.

Reply:
Removed `_is_comment(...)` in https://github.com/charmed-hpc/slurmutils/pull/15/commits/87e6fb2c690a86e6fccdd662acdf5806da72648c